addons/account/models/account_account.py

```diff
@@ -1053,193 +1055,278 @@ def get_import_templates(self):
             raise UserError(_(
                 "You do not have the right to perform this operation as you do not have access to the following companies: %s.",
                 ", ".join(c.name for c in forbidden_companies)
             ))
 
-        # Step 3: Update records in DB.
-        accounts_to_remove = self - account_to_merge_into
+        # Step 2: Create new accounts.
+        base_company = self.env.company if self.env.company in self.company_ids else self.company_ids[0]
+        companies_to_update = self.company_ids - base_company
+        check_company_fields = {fname for fname, field in self._fields.items() if field.relational and field.check_company}
+        new_account_by_company = {
+            company: self.copy(default={
+                'name': self.name,
+                'company_ids': [Command.set(company.ids)],
+                **{
+                    fname: self[fname].filtered(lambda record: record.company_id == company)
+                    for fname in check_company_fields
+                }
+            })
+            for company in companies_to_update
+        }
+        new_accounts = self.env['account.account'].union(*new_account_by_company.values())
 
-        # 3.1: Update foreign keys in DB
-        wiz = self.env['base.partner.merge.automatic.wizard'].new()
-        wiz._update_foreign_keys_generic('account.account', accounts_to_remove, account_to_merge_into)
+        # Step 3: Update foreign keys in DB.
+
+        new_account_id_by_company_id = {str(company.id): new_account.id for company, new_account in new_account_by_company.items()}
+        new_account_id_by_company_id[base_company.id] = self.id  # Needed for update of xmlids
+        new_account_id_by_company_id_json = json.dumps(new_account_id_by_company_id)
+        (self | new_accounts).invalidate_recordset()
+
+        # 3.1: Update fields on other models that reference account.account
+        many2x_fields = self.env['ir.model.fields'].search([
+            ('ttype', 'in', ('many2one', 'many2many')),
+            ('relation', '=', 'account.account'),
+            ('store', '=', True)
+        ])
+        for field_to_update in many2x_fields:
+            model = field_to_update.model
+            if not self.env[model]._auto:
+                continue
+            if not (query_company_id := _get_query_company_id(model)):
+                continue
+            if field_to_update.ttype == 'many2one':
+                table = self.env[model]._table
+                account_column = field_to_update.name
+                model_column = 'id'
+            else:
+                table = field_to_update.relation_table
+                account_column = field_to_update.column2
+                model_column = field_to_update.column1
+            records_to_update = self.env[model].search([(field_to_update.name, '=', self.id)])
+            records_to_update.invalidate_recordset([field_to_update.name])
```

Comment:
This will probably be a performance issue for little to no reason?
I don't think that splitting an account that was used in the same transaction as it was set on a record is a common case.
However, splitting an account that is used on millions of lines should be possible, no?

(It might be ok for property fields, and reference fields as these shouldn't touch too many records)

Reply:
I agree that the search + invalidation is completely overkill here. I've modified this to do a single `invalidate_all` before we start doing the queries.